### PR TITLE
Parameter wise non centered

### PR DESCRIPTION
**Pull request by AlexanderFengler**
This pull request introduces significant enhancements to the control of non-centered parameterization in hierarchical models, allowing users to specify non-centering at both the model and per-prior levels, including fine-grained control per distributional component. It also improves error handling and maintains backward compatibility, while adding comprehensive tests for these new behaviors.

**New features and parameterization control:**

* Added a `noncentered` keyword to `bmb.Prior` for per-prior control, allowing explicit override of the model-level default for non-centered parameterization on group-specific terms. (`bambi/priors/prior.py` [[1]](diffhunk://#diff-fd97ce397508f4ea8754452b93d54f28f4e96ad2049d48a52ba7594fdc0ab4f8R1-R269) [[2]](diffhunk://#diff-cc77c6f21d8a602d2442f00e14626aa266686a65855e558e080bb6d6e5dfae05R18-R34) [[3]](diffhunk://#diff-cc77c6f21d8a602d2442f00e14626aa266686a65855e558e080bb6d6e5dfae05R67-R68); `CHANGELOG.md` [[4]](diffhunk://#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR1-R11)
* Enhanced `Model.noncentered` to accept a `dict[str, bool]` for per-component defaults, enabling users to specify non-centering on a per-parameter basis (e.g., `{"mu": True, "sigma": False}`), with validation for unknown keys and sensible fallbacks for missing ones. (`bambi/models.py` [[1]](diffhunk://#diff-1aaef6370fa4f66c64e1552dc9b687aca1d1e7d9c55accee3d1bc8f8e3102c03L90-R98) [[2]](diffhunk://#diff-1aaef6370fa4f66c64e1552dc9b687aca1d1e7d9c55accee3d1bc8f8e3102c03R235-R258)
* Updated backend logic to resolve the effective non-centering setting for each group-specific term, prioritizing per-prior overrides, then component-specific defaults, then the historical model-wide default. (`bambi/backend/model_components.py` [[1]](diffhunk://#diff-56a40ec979b410fe1e38d67958d7118e8d6f93e79f93a3cd7a0b5d98053868feR145-R147); `bambi/backend/terms.py` [[2]](diffhunk://#diff-545d09cbb1bfe61852b88322d2e8db21ece072f350521f8a34bef41b5bf5090bL195-R198)

**Error handling and maintenance:**

* Relaxed the `NotImplementedError` for non-Normal group-specific priors: now, non-Normal priors with random hyperpriors are supported under the centered parameterization (`noncentered=False`), with improved error messages when unsupported configurations are requested. (`bambi/backend/terms.py` [[1]](diffhunk://#diff-545d09cbb1bfe61852b88322d2e8db21ece072f350521f8a34bef41b5bf5090bL207-R212); `CHANGELOG.md` [[2]](diffhunk://#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR1-R11)

**Testing and documentation:**

* Added extensive tests covering all combinations of per-prior, per-component, and model-level `noncentered` settings, including mixed and edge cases, as well as improved string representation and equality for the new `Prior.noncentered` field. (`tests/test_noncentered.py` [[1]](diffhunk://#diff-fd97ce397508f4ea8754452b93d54f28f4e96ad2049d48a52ba7594fdc0ab4f8R1-R269); `tests/test_priors.py` [[2]](diffhunk://#diff-58e209895f3d360ae93220119314c9af185ed71f24cdba465b27137a724734ffR129-R153)
* Updated documentation and changelog to reflect the new features and behaviors. (`CHANGELOG.md` [[1]](diffhunk://#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edR1-R11); `bambi/models.py` [[2]](diffhunk://#diff-1aaef6370fa4f66c64e1552dc9b687aca1d1e7d9c55accee3d1bc8f8e3102c03L90-R98); `bambi/priors/prior.py` [[3]](diffhunk://#diff-cc77c6f21d8a602d2442f00e14626aa266686a65855e558e080bb6d6e5dfae05R18-R34)

**Maintenance:**

* Minor updates to dependency configuration for `nutpie` in `pyproject.toml` to avoid duplicate declarations. (`pyproject.toml` [pyproject.tomlL117-R121](diffhunk://#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711L117-R121))

### CODE REVIEW

File: bambi/backend/terms.py [hunk 1]
```diff
@@ -204,7 +206,8 @@ def build_distribution(self, prior, label, dims=None):
                 return pm.Deterministic(label, offset * sigma, dims=dims)
 
             raise NotImplementedError(
-                "The non-centered parametrization is only supported for Normal priors"
+                f"Non-centered parametrization is only supported for Normal priors; "
+                f"got '{prior.name}'. Set noncentered=False to use the centered parametrization."
```
Comment (tomicapretto) [inline bambi/backend/terms.py]:
```suggestion
               f"The non-centered parametrization is only supported for Normal priors, got {prior.name}."
```

Users could have specified noncentered in several ways

File: bambi/models.py [hunk 2]
```diff
@@ -87,9 +87,12 @@ class Model:
         If `True` (default), priors are automatically rescaled to the data
         (to be weakly informative) any time default priors are used. Note that any priors
         explicitly set by the user will always take precedence over default priors.
-    noncentered : bool, optional
-        If `True` (default), uses a non-centered parameterization for normal hyperpriors on
-        grouped parameters. If `False`, naive (centered) parameterization is used.
+    noncentered : bool or dict[str, bool], optional
+        Default parameterization for group-specific terms with random hyperpriors.
```
Comment (tomicapretto) [inline bambi/models.py]:
```suggestion
        Default parameterization for group-specific terms.
```

File: pyproject.toml [hunk 3]
```diff
@@ -114,9 +114,11 @@ numpyro = ">=0.19,<0.20"
...
 numba = ">=0.61.2"
 # nutpie = ">=0.15,<0.16"
-
-[tool.pixi.feature.nutpie.pypi-dependencies]
-nutpie = { git = "https://github.com/pymc-devs/nutpie.git" }
+# NOTE: `nutpie @ git+...` is already declared in [project.optional-dependencies],
+# which pixi auto-imports as a feature. Re-declaring it here triggers
+# "nutpie is already a dependency" on `pixi install`. Left commented for now.
+# [tool.pixi.feature.nutpie.pypi-dependencies]
+# nutpie = { git = "https://github.com/pymc-devs/nutpie.git" }
```
Comment (tomicapretto) [inline pyproject.toml]: Rebase to main and this will go away I think

File: tests/test_noncentered.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    assert "continuous2|binary_cat_offset" in _named_vars(model)
+    assert "1|binary_cat_offset" not in _named_vars(model)
+
+
+def test_per_prior_false_beats_model_true(data_random_n100):
+    priors = {"continuous2|binary_cat": _hyper_normal(noncentered=False)}
+    model = bmb.Model(
+        "continuous1 ~ continuous2 + (continuous2|binary_cat)",
+        data_random_n100,
+        priors=priors,
```
Comment (tomicapretto) [inline tests/test_noncentered.py:38]: set noncentered=True to not rely on the default?

File: tests/test_priors.py [hunk 5]
```diff
@@ -126,6 +126,31 @@ def test_prior_eq():
     assert prior1 != "bmb.Prior"
 
 
+def test_prior_noncentered_field():
+    # Default is None (inherit Model.noncentered)
```
Comment (tomicapretto) [inline tests/test_priors.py]:
```suggestion
    # Default is None
```

**Review by tomicapretto — CHANGES_REQUESTED** [review]
Asking for a rebase and a few small changes, all the rest look good

File: docs/notebooks/per_parameter_noncentered.ipynb [hunk 6]
```diff
@@ -0,0 +1,649 @@
+{
```
Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p>Could you show the model graph instead? Like <code>m_nc.graph()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278175845'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p>Same as above</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278175850'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p><code>noncentered</code> without the <code>=</code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278175855'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p>Same as before with the graph. Also, can you directly write the priors in the <code>priors</code> dict to avoid the lambda function abstraction? I think a direct implementation can be easier for beginners. </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278175857'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p>A few asks:</p><ul><li>Could you separate the model fit from the display of summaries?</li><li>Can you increase number of chains to 4 and leave tune and draws equal to their defaults?</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278175859'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Comment by tomicapretto** [comment]
@AlexanderFengler, in order for the notebook to be displayed, you need to add it to the gallery in docs/notebooks/gallery.yml. I'm not sure what's the best category for this notebook, but feel free to create one

File: docs/notebooks/per_parameter_noncentered.ipynb (hunk 6 above)
Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p>Could you show the model graph instead? Like <code>m_nc.graph()</code></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278186176'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p>Same as above</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278186179'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p><code>noncentered</code> without the <code>=</code> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278186187'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p>Same as before with the graph. Also, can you directly write the priors in the <code>priors</code> dict to avoid the lambda function abstraction? I think a direct implementation can be easier for beginners. </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278186190'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (tomicapretto) [inline docs/notebooks/per_parameter_noncentered.ipynb:1]: <p>A few asks:</p><ul><li>Could you separate the model fit from the display of summaries?</li><li>Can you increase number of chains to 4 and leave tune and draws equal to their defaults?</li></ul><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/bambinos/bambi/pull/983/discussion/#comment-3278186195'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Review by tomicapretto — APPROVED** [review]
